crates/storage/src/store/multistore.rs

```diff
@@ -0,0 +1,69 @@
+use std::sync::Arc;
+
+use super::substore::SubstoreConfig;
+
+/// A collection of substore, each with a unique prefix.
+#[derive(Debug, Clone)]
+pub struct Multistore {
+    transparent_store: Arc<SubstoreConfig>,
+    substores: Vec<Arc<SubstoreConfig>>,
+}
+
+impl Multistore {
+    pub fn new(mut substores: Vec<Arc<SubstoreConfig>>) -> Self {
+        assert!(
+            !substores.is_empty(),
+            "multistore must have at least one substore"
+        );
```

Comment:
Why do we want to require this?

For that matter, why do we have a constructor at all -- since the `MultistoreConfig` is set on storage configuration, we should never be creating more than one, so it can be created once just with struct initialization in the part of the code that loads the configuration.